File: SQL Server 2012/Projects/C#/Chapter 23/ApClrObjects/ApClrObjects/Stored Procedures/InsertInvoiceTest.cs
//------------------------------------------------------------------------------
// <copyright file="CSSqlStoredProcedure.cs" company="Microsoft">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//------------------------------------------------------------------------------
using System;
using System.Data;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

public partial class StoredProcedures
{
    [Microsoft.SqlServer.Server.SqlProcedure]
    public static SqlInt32 InsertInvoiceTest(SqlString invoiceNumber)
    {
        // validate the invoiceNumber parameter
        if (invoiceNumber.IsNull || invoiceNumber == "")
        {
            throw new ApplicationException("Invalid InvoiceNumber. " +
                "You must specify an InvoiceNumber.");
        }

        // create and return a test invoice ID
        int newInvoiceID = 115;
        return (SqlInt32)newInvoiceID;
    }
}

File: SQL Server 2012/Projects/Visual Basic/Chapter 22/ApClrObjects/Stored Procedures/InsertInvoiceTest.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class StoredProcedures
    <Microsoft.SqlServer.Server.SqlProcedure()> _
    Public Shared Function InsertInvoiceTest(ByVal invoiceNumber As SqlString) _
            As SqlInt32
        ' Validate the invoiceNumber parameter
        If invoiceNumber.IsNull OrElse invoiceNumber = "" Then
            Throw New ApplicationException("Invalid invoice number. " &
                "You must specify an invoice number.")
        Else
            ' Create and return a test invoice ID
            Dim newInvoiceID As Integer = 115
            Return CType(newInvoiceID, SqlInt32)
        End If
    End Function
End Class
